modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultApplicationPolicyInstance.java

```diff
@@ -90,11 +97,26 @@ private void initPolicyContext() throws InitialisationException {
                                                                                      artifactPlugins,
                                                                                      new DefaultExtensionManagerFactory()));
 
-    artifactBuilder.withServiceConfigurator(customizationService -> customizationService
-        .overrideDefaultServiceImpl(MuleProperties.OBJECT_POLICY_MANAGER_STATE_HANDLER,
-                                    application.getMuleContext().getRegistry()
-                                        .lookupObject(MuleProperties.OBJECT_POLICY_MANAGER_STATE_HANDLER)));
-
+    artifactBuilder.withServiceConfigurator(customizationService -> {
+      MuleRegistry applicationRegistry = application.getMuleContext().getRegistry();
+      customizationService.overrideDefaultServiceImpl(OBJECT_LOCK_PROVIDER,
```

Comment:
Why don't you use createLifecycleFilterProxy for these objects?

Reply:
The @Inject in our extension failed if I proxied them. I checked though, and no dispose method was executed on those without the proxy so the policy lifecycle shouldn't be altering anything from app's context.
Nevertheless, do you see a way to overcome the @Inject error so that we can proxy everything?

Reply:
Can you add more context about why the extension failed? If failed because there was no available object to inject, then you probably you need to add another interface to the proxy in order to satisfy that requirement